keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -725,6 +725,21 @@ public double determinant() {
         return new LUDecomposition(matrix).getDeterminant();
     }
 
+    @Override
+    public double product() {
+        if (this.isScalar()) {
+            return this.scalar();
+        } else if (this.isVector()) {
+            if (this.getShape()[0] > 1) {
+                return tensor.prod(0).getDouble(0);
```

Comment:
Is there not a prod() that would work for both types of vectors? Can we make it work for arbitrarily ranked tensors as well?

Reply:
Nice, added the zero constructor prod which works for all tensor shapes